### PR TITLE
Closes #53: reapprove immediately after push

### LINKED ISSUE #53: Re-approvals only applied after successful CI run
There is a comment related to this within `marge/job.py`, including for context:
```
# Re-approve the merge request, in case us pushing it has removed
# approvals. Note that there is a bit of a race; effectively
# approval can't be withdrawn after we've pushed (resetting
# approvals) and CI runs.
```

Occasionally CI may fail due to transient network issues that are unrelated to the change made. In this case, Marge will error out and not bother attempting to reapply any approvals. GitLab doesn't remove approvals on CI failure, so it doesn't quite make sense that this happens with Marge.

This also applies to any potential exception that might occur between the force push and applying approvals, we need to restart marge and then manually approve again.

I'm unaware as to whether there is a historical reason for why approvals are reapplied when they are, but could they no be applied immediately after the rebase?

### PR DESCRIPTION
**Pull request by aschmolck**
Rather than waiting for CI to pass, just approve right away; this is more
convenient for transient errors such as flaky builds, and I see no obvious
downside.

### CODE REVIEW

File: marge/job.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def execute(self):
             log.exception('Unexpected Git error')
             merge_request.comment('Something seems broken on my local git repo; check my logs!')
             raise
-        except Exception:
+        except Exception as _ex:
```
Comment (jcpetruzza) [inline marge/job.py]: @aschmolk Why this change?

File: tests/test_job.py [hunk 2]
```diff
@@ -90,8 +90,13 @@ def __init__(self, gitlab_url=None):
...
         api.add_transition(
            GET('/projects/1234/repository/branches/useless_new_feature'),
+           Ok({'commit': _commit(id=rewritten_sha, status='running')}),
+            from_state='pushed',
+        )
+        api.add_transition(
+           GET('/projects/1234/repository/branches/useless_new_feature'),
            Ok({'commit': _commit(id=rewritten_sha, status='success')}),
-           from_state='passed'
+            from_state='passed'
```
Comment (jcpetruzza) [inline tests/test_job.py]: @aschmolck Extra space before `from_state` (also in the transition above)

File: marge/job.py (hunk 1 above)
Comment (aschmolck) [reply marge/job.py]: Just for debugging, it's convenient to have the exception around.

File: tests/test_job.py (hunk 2 above)
Comment (aschmolck) [reply tests/test_job.py]: Weird, that was due to a bug in emacs python mode indentation logic!
